SDWebImage/SDWebImageDownloader.m

```diff
@@ -308,6 +308,15 @@ - (nullable SDWebImageDownloadToken *)downloadImageWithURL:(nullable NSURL *)url
         [self.downloadQueue addOperation:operation];
     }
     UNLOCK(self.operationsLock);
+    
+    // Prioritize the operation queue priority when the current operation is pending in queue
+    if ((options & SDWebImageDownloaderHighPriority || options & SDWebImageDownloaderLowPriority) && !operation.isFinished && !operation.isExecuting) {
```

Comment:
Emm, we don't need do this check for all operations, only for operation already in operationQueue and isReady.

Reply:
This is check for that `the operation is pending on the queue`. What do you mean ?

Reply:
The combine usage `!operation.isFinished && !operation.isExecuting` is necessary for checking `operation is ready but not started`. So need this check.

The `options & SDWebImageDownloaderHighPriority` on the left is because of we only need to touch in this condition. This can slightly increase the performance because `isExecuting && isFinished` need synchronization inside its getter method.

The goal of this line, is to check `whether current new operation, or an operation is already in the queue and been re-queried, should update its queue priority`. So I think it's the right place.